src/MCPServer/lib/RPCServer.py

```diff
@@ -15,74 +17,116 @@
+        return fn(*args, **kwargs)
+    return wrap
+
+
+def pickle_result(fn):
+    """Worker handler decorator to pickle the returned value."""
+    def wrap(*args, **kwargs):
+        return cPickle.dumps(fn(*args, **kwargs))
+    return wrap
+
+
+@pickle_result
+@unpickle_payload
+@log_exceptions(LOGGER)
+def job_approve_handler(*args, **kwargs):
+    payload = kwargs['payload']
+    job_id = payload["jobUUID"]
+    chain = payload["chain"]
+    user_id = str(payload["uid"])
+    LOGGER.debug("Approving: %s %s %s", job_id, chain, user_id)
+    if job_id in choicesAvailableForUnits:
+        choicesAvailableForUnits[job_id].proceedWithChoice(chain, user_id)
+    return "approving: ", job_id, chain
+
+
+@pickle_result
+@log_exceptions(LOGGER)
+def job_awaiting_approval_handler(*args):
+    ret = etree.Element('choicesAvailableForUnits')
+    for uuid, choice in choicesAvailableForUnits.items():
         ret.append(choice.xmlify())
     return etree.tostring(ret, pretty_print=True)
 
 
-def approveJob(jobUUID, chain, user_id):
-    LOGGER.debug("Approving: %s %s %s", jobUUID, chain, user_id)
-    if jobUUID in choicesAvailableForUnits:
-        choicesAvailableForUnits[jobUUID].proceedWithChoice(chain, user_id)
-    return "approving: ", jobUUID, chain
-
-
-def gearmanApproveJob(gearman_worker, gearman_job):
-    try:
-        data = cPickle.loads(gearman_job.data)
-        jobUUID = data["jobUUID"]
-        chain = data["chain"]
-        user_id = str(data["uid"])
-        return cPickle.dumps(approveJob(jobUUID, chain, user_id))
-    # catch OS errors
-    except Exception:
-        LOGGER.exception('Error approving job')
-        raise
-
-
-def gearmanGetJobsAwaitingApproval(gearman_worker, gearman_job):
-    try:
-        return cPickle.dumps(getJobsAwaitingApproval())
-    # catch OS errors
-    except Exception:
-        LOGGER.exception('Error getting jobs awaiting approval')
-        raise
+@pickle_result
+@unpickle_payload
+@log_exceptions(LOGGER)
+def package_create_handler(*args, **kwargs):
+    """Handle create package request."""
+    payload = kwargs['payload']
+    create_package_args = (
+        payload.get('name'),
+        payload.get('type'),
+        payload.get('accession'),
+        payload.get('path'),
+        payload.get('metadata_set_id'),
+    )
+    return create_package(*create_package_args).pk
 
 
 def startRPCServer():
     gm_worker = gearman.GearmanWorker([django_settings.GEARMAN_SERVER])
     hostID = gethostname() + "_MCPServer"
     gm_worker.set_client_id(hostID)
-    gm_worker.register_task("approveJob", gearmanApproveJob)
-    gm_worker.register_task("getJobsAwaitingApproval", gearmanGetJobsAwaitingApproval)
+
+    # The tasks registered in this worker should not block.
+    gm_worker.register_task("approveJob", job_approve_handler)
+    gm_worker.register_task("getJobsAwaitingApproval", job_awaiting_approval_handler)
+    gm_worker.register_task("packageCreate", package_create_handler)
```

Comment:
`packageCreate` is really the only endpoint I'm adding this time. But I took the chance to introduce some decorators that I think can be helpful.